subsys/bluetooth/host/l2cap.c

```diff
@@ -1256,6 +1252,22 @@ static void l2cap_chan_seg_sent(struct bt_conn *conn, void *user_data)
 	l2cap_chan_tx_resume(BT_L2CAP_LE_CHAN(chan));
 }
 
+static bool test_and_dec(atomic_t *target)
+{
+	atomic_t old_value, new_value;
+
+	do {
+		old_value = atomic_get(target);
+		if (!old_value) {
+			return false;
+		}
+
+		new_value = old_value - 1;
+	} while (atomic_cas(target, old_value, new_value) == 0);
```

Comment:
Minor: I would think while(!atomic_cas()); but I see that atomic_cas has different return values based on ifdefs, and that both versions are used elsewhere in the code.

Reply:
`while(!atomic_cas())` and `while (atomic_cas() == 0)` are the same. MISRA-C wants us to not treat integers as booleans, so the specific == 0 is added.